src/common/ManagedCommon/ThemeHelpers.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.Win32;
+
+namespace ManagedCommon
+{
+    // Based on https://stackoverflow.com/a/62811758/5001796
+    public static class ThemeHelpers
+    {
+        [DllImport("dwmapi.dll")]
+        private static extern int DwmSetWindowAttribute(IntPtr hwnd, int attr, ref int attrValue, int attrSize);
+
+        internal const string HKeyRoot = "HKEY_CURRENT_USER";
+        internal const string HkeyWindowsTheme = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Themes";
+        internal const string HkeyWindowsPersonalizeTheme = $@"{HkeyWindowsTheme}\Personalize";
+        internal const string HValueAppTheme = "AppsUseLightTheme";
+        internal const string HValueCurrentTheme = "CurrentTheme";
+
+        // based on https://stackoverflow.com/questions/51334674/how-to-detect-windows-10-light-dark-mode-in-win32-application
+        public static AppTheme GetAppTheme()
+        {
+            int value = (int)Registry.GetValue($"{HKeyRoot}\\{HkeyWindowsPersonalizeTheme}", HValueAppTheme, 1);
+            return (AppTheme)value;
+        }
+
+        public static Theme GetCurrentTheme()
+        {
+            string theme = (string)Registry.GetValue($"{HKeyRoot}\\{HkeyWindowsTheme}", HValueCurrentTheme, string.Empty);
+            theme = theme.Split('\\').Last().Split('.').First().ToString();
+
+            switch (theme)
+            {
+                case "hc1":
+                    return Theme.HighContrastOne;
+                case "hc2":
+                    return Theme.HighContrastTwo;
+                case "hcwhite":
+                    return Theme.HighContrastWhite;
+                case "hcblack":
+                    return Theme.HighContrastBlack;
```

Comment:
I don't believe this is the accepted way to determine whether HC mode is enabled.

Terminal uses this (approved by the window manager folks, same method XAML uses)

https://github.com/microsoft/terminal/blob/ed27737233714dea77877624d1beeb49e2ccd36e/src/cascadia/WindowsTerminal/icon.cpp#L14-L25

Reply:
I took this from another piece of PowerToys, but since I don't actually use it, I think I will just remove it.

Reply:
Removed!